scripts/files/tests/fs_test.py

```diff
@@ -89,6 +89,18 @@ def test_write_sidecars_one_found(capsys: CaptureFixture[str], subtests: SubTest
     rmtree(target)
 
 
+def test_write_all_in_order(setup: str) -> None:
```

Comment:
does this test fail before the change?

Reply:
I have updated the test as discussed to one that fails before the code change.
Example of `written_files` list before the change:
`['/tmp/tmpjdifie0u/5', '/tmp/tmpjdifie0u/0', '/tmp/tmpjdifie0u/3', '/tmp/tmpjdifie0u/1', '/tmp/tmpjdifie0u/7', '/tmp/tmpjdifie0u/2', '/tmp/tmpjdifie0u/9', '/tmp/tmpjdifie0u/4', '/tmp/tmpjdifie0u/6', '/tmp/tmpjdifie0u/8']`
and after:
`['/tmp/tmp7gjme7sj/0', '/tmp/tmp7gjme7sj/1', '/tmp/tmp7gjme7sj/2', '/tmp/tmp7gjme7sj/3', '/tmp/tmp7gjme7sj/4', '/tmp/tmp7gjme7sj/5', '/tmp/tmp7gjme7sj/6', '/tmp/tmp7gjme7sj/7', '/tmp/tmp7gjme7sj/8', '/tmp/tmp7gjme7sj/9']`